src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -74,16 +74,24 @@ private long computeCountUsingStaticScale()
         return getRowCountForScale(1);
     }
 
-    private long computeCountUsingLogScale(int scale)
+    private long computeCountUsingLogScale(double scale)
     {
         int scaleSlot = getScaleSlot(scale);
         long delta = getRowCountForScale(DEFINED_SCALES[scaleSlot]) - getRowCountForScale(DEFINED_SCALES[scaleSlot - 1]);
-        float floatOffset = (float) (scale - DEFINED_SCALES[scaleSlot - 1]) / (float) (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
+        double floatOffset = (scale - DEFINED_SCALES[scaleSlot - 1]) / (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
 
-        return (int) (floatOffset * (float) delta) + getRowCountForScale(DEFINED_SCALES[0]);
+        long baseRowCount = 0;
+        if (scale < 1.0) {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[0]);
+        }
+        else {
+            baseRowCount = getRowCountForScale(DEFINED_SCALES[1]);
+        }
+        long count = (long) (floatOffset * (float) delta) + baseRowCount;
```

Comment:
when you cast the long to an int, you're truncating first (which is definitely a bug, but was copied from the C generator with the rest of the bugs).  If it doesn't show up at any of the scales, then it doesn't matter, but you see why that could be relevant.